system/jlib/jthread.cpp

```diff
@@ -54,10 +54,13 @@ void addThreadTermFunc(ThreadTermFunc onTerm)
 
 void callThreadTerminationHooks(bool isPooled)
 {
+    std::vector<ThreadTermFunc> keepHooks;
     for (auto hook: threadTermHooks)
     {
-        (*hook)(isPooled);
+        if ((*hook)(isPooled) && isPooled)
+           keepHooks.push_back(hook);
     }
+    threadTermHooks = keepHooks;
```

Comment:
better to swap

Reply:
Given there's seldom more than a single element (in fact there's usually none at all) I'm not sure it makes a lot of difference.